src/main/java/classify/user/InputParsing.java

```diff
@@ -900,14 +900,21 @@ protected static LocalDate parseDateFromString(String string) {
 
     private static LocalDate invalidDatePath() {
         logger.log(Level.WARNING, "Invalid date format entered." + '\n');
-        Ui.printInvalidDateError();
+        Ui.printInvalidDateFormatError();
         return LocalDate.now().plusDays(2);
     }
 
     private static boolean isDateValid(LocalDate paymentDate) {
 
-        return paymentDate.isBefore(LocalDate.now().plusDays(1))
-                && paymentDate.isAfter(LocalDate.parse(EARLIER_POSSIBLE_DATE));
+        if (paymentDate.isAfter(LocalDate.now().plusDays(1)) || 
```

Comment:
Good way to check for date being in the valid range stipulated by a traditional OS.